arc/database/batch_processor.py

```
@@ -0,0 +1,386 @@
+        """Flush the outcomes batch to database."""
+        if not self._outcome_batch:
+            return
+        
+        batch_size = len(self._outcome_batch)
+        start_time = time.time()
+        
+        # Check circuit breaker
+        if self.circuit_breaker and not self.circuit_breaker.can_execute():
+            logger.warning(f"Circuit breaker OPEN - skipping batch of {batch_size} outcomes")
+            return
+        
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)